pypeit/edgetrace.py

```diff
@@ -3970,326 +3988,235 @@ def fully_masked_traces(self, flag=None, exclude=None):
+
+        xtop_mm = self.spectrograph.slitmask.top[:, 0]
+        xbot_mm = self.spectrograph.slitmask.bottom[:, 0]
+
+        ytop_mm = self.spectrograph.slitmask.top[:, 1]
+        ybot_mm = self.spectrograph.slitmask.bottom[:, 1]
+
+        slitindx = self.spectrograph.slitmask.slitindx
+        slit_id = self.spectrograph.slitmask.slitid
+
+        npoints = 250
+        nslits = self.spectrograph.slitmask.nslits
+        wave = np.arange(npoints) * 24. + 4000.
+
+        # Left (bottom) and right (top) traces in pixels from optical model
+        # bottom
+        xb_img, yb_img, ccd_b, xb_pix, yb_pix = self.spectrograph.mask_to_pixel_coordinates(x=xbot_mm, y=ybot_mm,
+                                                                                            wave=wave)
+        # top
+        xt_img, yt_img, ccd_t, xt_pix, yt_pix = self.spectrograph.mask_to_pixel_coordinates(x=xtop_mm, y=ytop_mm,
+                                                                                            wave=wave)
+
+        # Per each slit we take the median value of the left and right trace over the wavelength direction
+        ccdnum=self.traceimg.detector.det
+        model_xb = np.zeros(nslits)
+        model_xt = np.zeros(nslits)
+
+        for i in range(nslits):
+            model_xb[i] = -1 if xb_pix[i, ccd_b[i, :] == ccdnum].shape[0] == 0 else np.median(
+                                                                                    xb_pix[i, ccd_b[i, :] == ccdnum])
+            model_xt[i] = -1 if xt_pix[i, ccd_t[i, :] == ccdnum].shape[0] == 0 else np.median(
+                                                                                    xt_pix[i, ccd_t[i, :] == ccdnum])
+            whgood = np.where(xt_img[i, :] > -1e4)[0]
+            npt_x_img = int(whgood.shape[0] / 2.)
+            whgood = whgood[:npt_x_img] if ccdnum <= 4 else whgood[npt_x_img:]
+            if (model_xb[i] == -1) & (model_xt[i] >= 0):
+                model_xb[i] = model_xt[i] - np.median((xt_img - xb_img)[i, whgood])
+            if (model_xt[i] == -1) & (model_xb[i] >= 0):
+                model_xt[i] = model_xb[i] + np.median((xt_img - xb_img)[i, whgood])
+
+
+
+        if debug is True:
+            print("slitindex    length      width       xcen_mm         model.xb  model.xt")
+            for i in range(xcen_mm.shape[0]):
+                if (model_xb[sortindx][i] != -1) | (model_xt[sortindx][i] != -1):
+                    print("{}  {}  {}  {}   {}      {}  {}".format(slitindx[sortindx][i], slit_id[sortindx][i],
+                        self.spectrograph.slitmask.length[sortindx][i], self.spectrograph.slitmask.width[sortindx][i],
+                                                    xcen_mm[sortindx][i], model_xb[sortindx][i], model_xt[sortindx][i]))
+
+        reference_row = self.left_pca.reference_row if self.par['left_right_pca'] else self.pca.reference_row
+        xb_det = self.edge_fit[reference_row,self.is_left]
```

Comment:
Please change all uses of x and y in variable names in this routine to spat and spec.

Reply:
Done!